jf_agent/git/gitlab_adapter.py

```diff
@@ -428,42 +432,52 @@ def _normalize_commit(
-                created_at=note.created_at,
-                system_generated=note.system,
+    if merge_request.note_list:
+        try:
+            return [
+                NormalizedPullRequestComment(
+                    user=_normalize_user(note.author),
+                    body=sanitize_text(note.body, strip_text_content),
+                    created_at=note.created_at,
+                    system_generated=note.system,
+                )
+                for note in merge_request.note_list
+            ]
+        except (requests.exceptions.RetryError, gitlab.exceptions.GitlabHttpError) as e:
+            log_and_print_request_error(
+                e,
+                f'standardizing PR comments for merge_request {merge_request.id} -- '
+                f'handling it as if it has no comments',
             )
-            for note in merge_request.note_list
-        ]
-    except (requests.exceptions.RetryError, gitlab.exceptions.GitlabHttpError) as e:
-        log_and_print_request_error(
-            e,
-            f'standardizing PR comments for merge_request {merge_request.id} -- '
-            f'handling it as if it has no comments',
-        )
-        return []
+            return []
+    print(
+        f'\nNo comments found. Standardizing PR comments for merge_request ID: {merge_request.id} -- \n handling it as if it has no comments'
+    )
+    return []
 
 
 def _get_normalized_approvals(merge_request):
-    try:
-        return [
-            NormalizedPullRequestReview(
-                user=_normalize_user(approval['user']),
-                foreign_id=approval['user']['id'],
-                review_state='APPROVED',
+    if merge_request.approved_by:
+        try:
+            return [
+                NormalizedPullRequestReview(
+                    user=_normalize_user(approval['user']),
+                    foreign_id=approval['user']['id'],
+                    review_state='APPROVED',
+                )
+                for approval in merge_request.approved_by
+            ]
+        except (requests.exceptions.RetryError, gitlab.exceptions.GitlabHttpError) as e:
+            log_and_print_request_error(
+                e,
+                f'standardizing PR approvals for merge_request {merge_request.id} -- '
+                f'handling it as if it has no approvals',
             )
-            for approval in merge_request.approved_by
-        ]
-    except (requests.exceptions.RetryError, gitlab.exceptions.GitlabHttpError) as e:
-        log_and_print_request_error(
-            e,
-            f'standardizing PR approvals for merge_request {merge_request.id} -- '
-            f'handling it as if it has no approvals',
-        )
-        return []
+            return []
+    print(
+        f'\nNo approvals found. Standardizing PR approvals for merge_request {merge_request.id} -- \nhandling it as if it has no approvals'
+    )
+    return []
```

Comment:
Same thing here -- I think we can keep the old code, but just change this line:
```
for approval in merge_request.approved_by
```

To this:
```
for approval in merge_request.approved_by or []
```

Reply:
""